Library/Homebrew/tap.rb

```diff
@@ -280,4 +317,75 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
+
+  undef_method :install
+  undef_method :uninstall
+  undef_method :command_files
+  undef_method :pinned_symlink_path
+  undef_method :pinned?
+  undef_method :pin
+  undef_method :unpin
+  undef_method :to_hash
```

Comment:
I think that's better, yep, but I think the core issue is that currently the core formulae are not a tap. Maybe `SharedFormulaeContainer` or something to clarify that and it can be renamed if/when the core formulae are actually in a tap?